lib/engine_socketio.js

```diff
@@ -118,10 +118,10 @@ SocketIoEngine.prototype.step = function (requestSpec, ee) {
     return engineUtil.createLoopWithCount(requestSpec.count || -1, steps);
   }
 
-  let f = function(context, callback) {
+  let f = function(context, callback, socketio) {
```

Comment:
Ah you're right, I overlooked that part. I changed the code to do the following:

If a `emit` action is defined in requestSpec, it will validate that a namespace is defined, or set the default namespace ( `/` ) http://socket.io/docs/rooms-and-namespaces/#default-namespace. _( Maybe it's smart to add configurable defaults for any action somewhere. )_

`getSocket` has been renamed to `loadContextSocket`, this function no longer returns a socket but instead ensures that the namespace socket is loaded for the supplied context. 

The `f` function now gets the correct namespaced socket from the context based on requestSpec.emit.namespace.

Now the socketio parameter can be removed from the `f` function, and are the parameter equal for `preStep` and `f`

I think omitting preStep is adding unnecessary complexity. It's basically a decorator that ensures the socket for the specific namespace is loaded before continuing execution.